examples/crd_derive_schema.rs

```diff
@@ -19,15 +18,18 @@ use serde::{Deserialize, Serialize};
 // - https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting
 // - https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting-and-nullable
 
-#[derive(CustomResource, Serialize, Deserialize, Default, Debug, PartialEq, Eq, Clone, JsonSchema)]
+#[derive(CustomResource, ValidateSchema, Serialize, Deserialize, Default, Debug, PartialEq, Eq, Clone)]
 #[kube(
     group = "clux.dev",
     version = "v1",
     kind = "Foo",
     namespaced,
     derive = "PartialEq",
-    derive = "Default"
+    derive = "Default",
+    rule = Rule::new("self.metadata.name != 'forbidden'"),
 )]
+#[serde(rename_all = "camelCase")]
+#[cel_validate(rule = Rule::new("self.nonNullable == oldSelf.nonNullable"))]
```

Comment:
Yes, that’s the idea. As the type is generated, only way how `derive` can get to it is via the `kube` macro itself. Otherwise it gets assigned to the `Spec` structure. Which is great, since it allows to define structure specific rules, and there is no need to define them manually each time the structure is used in as a type for a field.